compiler/src/main/scala/org/scalajs/core/compiler/GenJSFiles.scala

```diff
@@ -29,6 +29,19 @@ trait GenJSFiles extends SubComponent { self: GenJSCode =>
     try {
       ir.InfoSerializers.serialize(output, Infos.generateClassInfo(tree))
       ir.Serializers.serialize(output, tree)
+    } catch {
+      case e: ir.InvalidIRException =>
+        e.tree match {
+          case ir.Trees.UndefinedParam() =>
+            reporter.error(NoPosition, "Found a dangling UndefinedParam at " +
```

Comment:
Consider using `sym.pos`. That would at least give an anchor for IDEs to display the error.